### DISCUSSION TITLE
unable to load files with fs.readFileSync

### DISCUSSION THREAD

**Discussion by forrest-lyman** [vercel/community#5953] [Help]
### Summary

I have a next.js app that is running fine locally, but can't load json files on vercel. 

<img width="2306" alt="image" src="https://github.com/vercel/community/assets/161376305/b67b0965-5150-48e4-80db-ff83cd1a6a83">

I have tried:

- fs.readFileSync(path.join(process.cwd(), 'src', 'content', 'indexes', 'tags.json'), 'utf-8')
- fs.readFileSync(path.join( 'src', 'content', 'indexes', 'tags.json'), 'utf-8')
- fs.readFileSync(path.join( '@/src', 'content', 'indexes', 'tags.json'), 'utf-8')

### Steps to Reproduce

https://culipedia.vercel.app/search/salmon

**Comment by forrest-lyman** [comment]
Note that I have confirmed that the files were deployed...

<img width="867" alt="image" src="https://github.com/vercel/community/assets/161376305/63718929-19ed-479a-a6ef-9cfad1abfaaf">

**Reply by BadRed** [reply]
Having the same issue - worked fine about a month ago with Next 13.x.
Trying to load the static file via server action.

**Comment by jacargentina** [comment]
Any solution to this? Having similar problem with a Remix app.

**Comment by sahilatahar** [comment]
If you try to use `fs.readFileSync` in a component that is rendered on the client-side, it will fail because `fs` is a Node.js module and is not available in the browser environment.
You should only use `fs.readFileSync` in server-side code, such as in `getStaticProps`, `getServerSideProps`, or API routes.

If you need to read a file at build time (for static generation), you can use `getStaticProps.`

```jsx
import fs from 'fs';
import path from 'path';

export async function getStaticProps() {
  const filePath = path.join(process.cwd(), 'src', 'content', 'indexes', 'tags.json');
  const fileContents = fs.readFileSync(filePath, 'utf-8');
  const data = JSON.parse(fileContents);

  return {
    props: {
      data
    }
  };
}

const PageComponent = ({ data }) => {
  return (
    <div>
      <pre>{JSON.stringify(data, null, 2)}</pre>
    </div>
  );
};

export default PageComponent;

```
